File: src/main/java/com/server/computerscience/chatbot/dto/request/ChatMessageDto.java
package com.server.computerscience.chatbot.dto.request;

import java.util.Collections;
import java.util.List;

import com.server.computerscience.chatbot.domain.ChatContentType;
import com.server.computerscience.chatbot.domain.ChatRole;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatMessageDto {
	private String role;
	private List<ChatContentDto> content;

	public static ChatMessageDto from(String text, ChatRole role) {
		
		return ChatMessageDto.builder()
			.role(role.getLower())
			.content(Collections.singletonList(ChatContentDto.from(ChatContentType.TEXT, text)))
			.build();
	}
}

File: common/src/main/java/com/server/computerscience/chatbot/dto/request/ChatMessageDto.kt
package com.server.computerscience.chatbot.dto.request

import com.server.computerscience.chatbot.domain.ChatContentType
import com.server.computerscience.chatbot.domain.ChatRole
import com.server.computerscience.chatbot.dto.request.ChatContentDto.Companion.from

class ChatMessageDto(
	val role: String,
	val content: List<ChatContentDto?>,
) {
	companion object {
		@JvmStatic
		fun from(
			text: String,
			role: ChatRole,
		): ChatMessageDto =
			ChatMessageDto(
				role = role.lower,
				content = listOf(from(ChatContentType.TEXT, text)),
			)

		@JvmStatic
		fun from(
			chatContents: List<ChatContentDto?>,
			role: ChatRole,
		): ChatMessageDto =
			ChatMessageDto(
				role = role.lower,
				content = chatContents,
			)
	}
}
